### ISSUE TITLE
rowserrcheck false positive when deferred Close() is wrapped in a func

### ISSUE THREAD

**Issue by stevendanna** [jingyugao/rowserrcheck#16]
rowserrcheck appears to produce a false positive when a deferred rows.Close() is wrapped in a func. This is perhaps similar to #11 , but occurs even on versions that should include that fix.

The false positive can be reproduced by the following example:

```
> nl -ba repro.go 
     1  package foo
     2  
     3  import (
     4          "database/sql"
     5          "fmt"
     6  )
     7  
     8  func Bug(db *sql.DB) {
     9          rows, err := db.Query("select 1")
    10          if err != nil {
    11                  panic(err)
    12          }
    13  
    14          defer func() { _ = rows.Close() }()
    15          for rows.Next() {
    16                  fmt.Println("new rows")
    17          }
    18  
    19          if err := rows.Err(); err != nil {
    20                  panic(err)
    21          }
    22  }
> go vet -vettool $(which rowserrcheck) ./repro.go 
# command-line-arguments
./repro.go:9:23: rows.Err must be checked
```

Interestingly, if `defer func() { _ = rows.Close() }()` at line 14 is changed to `defer rows.Close()`, the false positive goes away.

<details><summary>reproduction file</summary>

```go
package foo

import (
        "database/sql"
        "fmt"
)

func Bug(db *sql.DB) {
        rows, err := db.Query("select 1")
        if err != nil {
                panic(err)
        }

        defer func() { _ = rows.Close() }()
        for rows.Next() {
                fmt.Println("new rows")
        }

        if err := rows.Err(); err != nil {
                panic(err)
        }
}
```
</details>

**Comment by jingyugao** [comment]
thank you for report this issue.I will fix it recently
